app/src/main/java/ru/yandex/architectureproject/presentation/viewmodel/TaskViewModel.kt

```diff
@@ -30,12 +32,29 @@ class TaskViewModel(
     private val _state = MutableStateFlow<TaskState>(TaskState.Loading)
     val state: StateFlow<TaskState> = _state.asStateFlow()
 
+    private val taskForDeletionJobMap: MutableMap<Int, Job?> = mutableMapOf()
+
     init {
         reduce(TaskAction.LoadTasks)
     }
 
     fun reduce(action: TaskAction) {
-        // TODO: Здесь должна быть обработка действий
+        viewModelScope.launch {
+            when(action) {
+                is TaskAction.LoadTasks -> loadTasks()
+                is TaskAction.AddTask -> addTaskUseCase(action.task)
+                is TaskAction.UpdateTaskStatus -> {
+                    if (action.isDone) {
+                        taskForDeletionJobMap[action.taskId] = this.coroutineContext.job
+                        completeTaskUseCase(action.taskId)
+                    } else {
+                        taskForDeletionJobMap[action.taskId]?.cancel()
```

Comment:
Помимо отмены джобы её лучше также удалить из мапы, чтобы сборщик мусора мог удалить эту джобу из памяти:
```suggestion
                        taskForDeletionJobMap[action.taskId]?.remove()?.cancel()
```